extension/encoding/cloudwatchencoding/config.go

```diff
@@ -0,0 +1,31 @@
+package cloudwatchencoding
+
+import (
+	"fmt"
+	"go.opentelemetry.io/collector/component"
+)
+
+type contentEncoding string
+
+const (
+	NoEncoding  = ""
+	GZipEncoded = "gzip"
+)
+
+type Config struct {
+	Encoding contentEncoding `mapstructure:"content_encoding"`
+}
+
+func createDefaultConfig() component.Config {
+	return &Config{}
+}
+
+func (c *Config) Validate() error {
+	switch c.Encoding {
+	case NoEncoding:
+	case GZipEncoded:
+	default:
+		return fmt.Errorf("unknown content encoding %s", c.Encoding)
```

Comment:
For highlighting what a user has entered, I suggest using `%q` instead of `%s` since it will wrap the string into quotes making it clearer on what the value is.

Reply:
Love this, thanks!